util/dvsim/SGE.py

```diff
@@ -0,0 +1,337 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+# ----------------------------------
+# SGE.py
+#       _JobData Class
+# ----------------------------------
+import logging
+import pwd
+import os
+import re
+import subprocess
+import time
+
+#
+from qsubopts import qsubOptions
+sgeJob_t = qsubOptions()
+
+
+class _JobData:
+    """
+    Internal helper class to manage job data from qstat
+    """
+    def __init__(self, qstat_job_line):
+        # The format of the line goes like
+        # job-ID prior name  user  state submit/start at queue slots ja-task-ID
+
+        tokens = qstat_job_line.split()
+
+        assert len(tokens) >= 8, 'Not a valid qstat line: ' + qstat_job_line
+        # Job line must have at least 8 tokens
+        self.id = int(tokens[0])
```

Comment:
Notice this and the following line also can throw a ValueError.

Reply:
I don't see how this and the other issue I flagged , did you upload your changes?

Reply:
As I see , Sharon still didn’t upload my changes . I’m going to contact with her